src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,107 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 1 << 4;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int NULL_INDEX = 0;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size == table.length * DEFAULT_LOAD_FACTOR) {
+            sizeUp();
+        }
+        Node<K, V> newNode = new Node<>(Math.abs(Objects.hash(key)), key, value, null);
+        int positionInArray = Math.abs(Objects.hash(key) % table.length);
```

Comment:
```suggestion
        int index = getIndex();
```
create method getIndex()